src/qt/walletcontroller.cpp

```diff
@@ -308,6 +308,6 @@ void OpenWalletActivity::open(const std::string& path)
 
         if (wallet) m_wallet_model = m_wallet_controller->getOrCreateWallet(std::move(wallet));
 
-        QTimer::singleShot(0, this, &OpenWalletActivity::finish);
+        QMetaObject::invokeMethod(this, "finish", Qt::QueuedConnection);
```

Comment:
What is the idea? AFAIK `singleShot` does a queued connection too.

Reply:
> What is the idea? AFAIK `singleShot` does a queued connection too.

I thought so too. But careful reading https://doc.qt.io/qt-5/qtimer.html didn't point a such statement.

Reply:
Also I'm curious why only windows reveals this bug?